### ISSUE TITLE
Sampled softmax loss support

### ISSUE THREAD

**Issue by okuchaiev** [google/seq2seq#147]
Please consider adding sampled softmax loss, in addition to "cross_entropy_sequence_loss".
For tasks with large target vocabularies, the speedup can be significant (with, perhaps, minor accuracy loss per step).
Even on "nmt_large" config with batch size of 128 and voc size of 32,000 I am getting about **x1.224** speedup.
It is, however, a little tricky to add. I have a first draft here: https://github.com/okuchaiev/seq2seq/tree/sampled_softmax_first_try 
It is not ready to be merged yet.
Let me know if this is of interest - I plan to polish my implementation and would appreciate feedback on doing it right.

**Comment by sefialbo** [comment]
is it ready to use ? this feature is really helpful
